File: src/Samples/SharpPie/src/SharPie/Controls/UxListBox.php
<?php

namespace SharPie\Controls;
use Peachpie\Avalonia\Experimental\ListBoxWrapper;

/**
 * @property $Items
 * @property $Control
 */
class UxListBox extends ListBoxWrapper
{
    public function __get($name)
    {
        echo "Получение '$name'\n";
        return $this->GetProperty($name);
    }

    public function __set($name , $value) {
        echo "Установка '$name'\n";
        $this->SetProperty($name);
    }
}

File: src/Peachpie.Avalonia/Experimental/UxListBox.cs
﻿using System.Collections;
using Avalonia.Controls;
using Pchp.Core;
using System.Collections.Generic;

namespace Peachpie.Avalonia.Experimental;

[PhpType]
public class UxListBox : BaseWrapper<ListBox>
{
    public PhpArray Items
    {
        set => Control.Items = ConvertItemsToListBoxItems(value);
        get => ConvertListBoxItemsToItems(Control.Items);
    }
        
    private IEnumerable ConvertItemsToListBoxItems(PhpArray items)
    {
        var listBoxItems = new List<object>();
        foreach (var item in items)
        {
            var listBoxItem = new ListBoxItem();
            listBoxItem.Content = PhpValue.FromClass(item.Value);
            listBoxItems.Add(listBoxItem);
        }

        return listBoxItems;
    }
        
    private PhpArray ConvertListBoxItemsToItems(IEnumerable listBoxItems)
    {
        var items = new PhpArray();
        foreach (var listBoxItem in listBoxItems)
        {
            items.Add(PhpValue.FromClass(listBoxItem));
        }

        return items;
    }
}